specification/compute/resource-manager/Microsoft.Compute/preview/2021-06-01-preview/diagnostic.json

```diff
@@ -206,6 +206,98 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection/registerStorageConfiguration": {
+      "post": {
+        "tags": [
+          "registerStorageConfiguration"
+        ],
+        "operationId": "DiskInspectionStorageConfiguration_Register",
+        "x-ms-long-running-operation": false,        
+        "description": "Register a storageAccount for a subscription used for DiskInspection",
+        "consumes": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "storageConfigurationInput",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/StorageConfigurationInput"
+            },
+            "description": "Storage Configuration object supplied in the body of the Post to cache storageAccount for a given subscription."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Succeeded"          
+          },
+          "default": {
+            "description": "Error response describing why the register operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create a request for registering a storageAccount information.": {
+            "$ref": "./examples/RegisterStorageConfiguration.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection/validateStorageConfiguration": {
```

Comment:
"registering" a storage config appears to only register it for a single resource (the "resourceId" property). What is "validate" actually validating?  That ANY resource in the subscription and location has a storage config?

Reply:
After reviewing the existing functionality, we have agreed to not have "resourceId" information passed in as part of "registerStorageConfiguration" API. The "registerStorageConfiguration" API is intended to create a configuration entry between **subscriptionId -> storageAccount (SubscriptionLevel)**. "ValidateStorageConfiguration" will lookup if such a configuration exists for a subscription and perfom RBAC checks on SA for caller before sending response. I will update swagger and send out another iteration with the change.